bootstrap/src/index.ts

```diff
@@ -65,12 +65,22 @@ const executeSync = (execute: Execute): ExecuteSync => {
   // const initMiddleware = withMiddleware(execute)
 
   // Return sync function
-  return (data: AdapterRequest, callback: any) => {
+  return async (data: AdapterRequest, callback: any) => {
     // We init on every call because of cache connection broken state issue
-    return withMiddleware(execute)
-      .then((executeWithMiddleware) => executeWithMiddleware(data))
-      .then((result) => callback(result.statusCode, result))
-      .catch((error) => callback(error.statusCode || 500, Requester.errored(data.id, error)))
+    try {
+      const executeWithMiddleware = await withMiddleware(execute)
+      const result = await executeWithMiddleware(data)
+      // only consider registering a warmup request if the original one was successful
+      // and we have caching enabled
+      if (util.parseBool(process.env.CACHE_ENABLED)) {
+        store.dispatch(
+          actions.warmupRequestSubscribed({ data, executeFn: executeWithMiddleware, id: data.id }),
```

Comment:
Yeah, i think that's a good idea, what do you think about just keeping it more coupled for now, just for LOC sake? We're moving off of this soon anyway, but this would be good to refactor out when we do our rewrite